main/src/ca/uwaterloo/flix/language/ast/TypeConstructor.scala

```diff
@@ -414,6 +414,6 @@ object TypeConstructor {
   /**
     * A type constructor which represents an erroneous type of the given `kind`.
     */
-  case class Error(kind: Kind) extends TypeConstructor
+  case class Error(id: Int, kind: Kind) extends TypeConstructor
```

Comment:
What's the id for?

Reply:
In `EffUnification2` I map error effects to solver variables. I need different errors to have different IDs, otherwise they all get mapped to the same variable, which causes confusing errors.

Reply:
Could we do this in its own PR to make this easier to digest?